calico/felix/fetcd.py

```diff
@@ -418,7 +423,7 @@ def on_orch_delete(self, response, hostname, orchestrator):
         _log.info("Orchestrator dir %s/%s deleted, removing contained hosts",
                   hostname, orchestrator)
         for endpoint_id in list(self.endpoint_ids_per_host[hostname]):
-            if endpoint_id.orchestrator == orchestrator:
+            if endpoint_id.orchestrator == orchestrator.encode("utf8"):
```

Comment:
Sorry - I seem to be picking lots of nits today. More efficient to encode orchestrator once at the top.